vsock/src/main.rs

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+mod rxops;
+mod rxqueue;
+mod thread_backend;
+mod txbuf;
+mod vhu_vsock;
+mod vhu_vsock_thread;
+mod vsock_conn;
+
+use clap::Parser;
+use std::{
+    convert::TryFrom,
+    process,
+    sync::{Arc, RwLock},
+};
+use vhost::{vhost_user, vhost_user::Listener};
+use vhost_user_backend::VhostUserDaemon;
+use vhu_vsock::{VhostUserVsockBackend, VsockArgs, VsockConfig};
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+/// This is the public API through which an external program starts the
+/// vhost-user-vsock backend server.
+pub(crate) fn start_backend_server(vsock_config: VsockConfig) {
+    loop {
+        let vsock_backend = Arc::new(RwLock::new(
+            VhostUserVsockBackend::new(vsock_config.clone()).unwrap(),
+        ));
+
+        let listener = Listener::new(vsock_config.get_socket_path(), true).unwrap();
+
+        let mut vsock_daemon = VhostUserDaemon::new(
+            String::from("vhost-user-vsock"),
+            vsock_backend.clone(),
+            GuestMemoryAtomic::new(GuestMemoryMmap::new()),
+        )
+        .unwrap();
+
+        let mut vring_workers = vsock_daemon.get_epoll_handlers();
+
+        if vring_workers.len() != vsock_backend.read().unwrap().threads.len() {
+            println!("Number of vring workers must be identical to number of backend threads");
```

Comment:
Should we do something here instead of just notify the user? Can we use error!() here for reporting?

Reply:
I guess it was here just for debug purpose, so I would remove it or just add an assert!().
VhostUserDaemon takes this information from `VhostUserVsockBackend::queues_per_thread`.
In this case `VhostUserVsockBackend` setup a single thread and a single element in `queues_per_thread`, so this check should be always false.

Reply:
Or maybe I can add a test in main.c to check these things.